### PR TITLE
Automate Credential Adding Process

### PR DESCRIPTION
**Pull request by michaelyoungstrom**
Automate uploading credentials to Jenkins by defining them in a yaml file .

cc: @edx/devops

### CODE REVIEW

File: src/main/groovy/6importCredentials.groovy [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+            cred = new UsernamePasswordCredentialsImpl(
+                scope,
+                id,
+                description,
+                username,
+                password
+            )
+            credentialsStore.addCredentials(
+                globalDomain,
```
Comment (estute) [inline src/main/groovy/6importCredentials.groovy]: Can you explain how scope differs from domain? If you create a user or system-scoped credential, do you still put it into the 'global' domain?

File: src/main/groovy/6importCredentials.groovy [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                Secret.fromString(text)
+            )
+            credentialsStore.addCredentials(globalDomain, secretText)
+            break
+        case 'secretFile':
+            path = newCredential.path
+            fileName = newCredential.name
+
+            // Make sure the path and name is not empty
+            if (!path || !fileName) {
```
Comment (estute) [inline src/main/groovy/6importCredentials.groovy]: I think you should also check that the file exists, or wrap the new file + copy in a try/catch

File: src/main/groovy/6importCredentials.groovy (hunk 1 above)
Comment (michaelyoungstrom) [inline src/main/groovy/6importCredentials.groovy]: I believe Domain is more for just organizing the credentials. While the scope actually affects where they can be used.
```

System
    This credential is only available to the object on which the credential is associated. Typically you would use system-scoped credentials for things like email auth, slave connection, etc, i.e. where the Jenkins instance itself is using the credential. Unlike the global scope, this significantly restricts where the credential can be used, thereby providing a higher degree of confidentiality to the credential. 
Global
    This credential is available to the object on which the credential is associated and all objects that are children of that object. Typically you would use global-scoped credentials for things that are needed by jobs.
```

I defaulted Domain to Global as thats what we do in build jenkins. Perhaps I should add support to creating new domains? Also it looks like User isn't supported so I'm going to remove it

File: src/main/groovy/6importCredentials.groovy (hunk 2 above)
Comment (michaelyoungstrom) [inline src/main/groovy/6importCredentials.groovy]: Addressed.

**Comment by estute** [comment]
Thanks for the updates @michaelyoungstrom

**Pull request merged**